### PR TITLE
GHA: auto-sync noetic-devel to master

### PR DESCRIPTION
**Pull request by rhaschke**
Added new job to automatically fast-forward push noetic-devel to master after a successful push of master on ros-planning. 
This will automate the process of keeping master and noetic-devel in sync.

Note, the job will not run on pull-requests, different branches than master being pushed, or on forked repos.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -9,7 +9,6 @@ on:
   push:
     branches:
       - master
-      - "[kmn]*-devel"
```
Comment (v4hn) [inline .github/workflows/ci.yaml:12]: I see why you would you disable the workflow for noetic, but why do you not want to run this on k/m pushes anymore?
Even if the pull-requests will be tested independently I would also test pushes.

Comment (rhaschke) [reply .github/workflows/ci.yaml:12]: This file _is part of_ the `master` branch. The kinetic-devel and melodic-devel branches have their own CI config.
Thus, the setting was non-sense from the very beginning. However, it allowed to easily compare/share the config between those branches :wink:

**Comment by v4hn** [comment]
This action currently fails because of the protected branch. Is there some way to allow the action to push while everything else should still require the review?

> remote: error: GH006: Protected branch update failed for refs/heads/noetic-devel.        
> remote: error: Required status check "WIP" is expected. At least 1 approving review is required by reviewers with write access.        
> To https://github.com/ros-planning/moveit
> ! [remote rejected] master -> noetic-devel (protected branch hook declined)

**Comment by rhaschke** [comment]
I have seen that :disappointed: The problem didn't show up in my tests, unfortunately.
I see two options:
- Disable required checks for noetic-devel branch (but keep them for master, of course).
- Setup a secret token to push as rhaschke, who has correct permissions (preferred).

**Comment by rhaschke** [comment]
Looks like my suggested change [still doesn't work](https://github.com/ros-planning/moveit/runs/4140416981). The only way is to [relax required checks](https://github.com/semantic-release/github/issues/175#issuecomment-891108176).

**Pull request merged**